utilities/bulk-rename.py

```diff
@@ -0,0 +1,179 @@
+#! /usr/bin/env python
+""" Tool to replace strings in NIPAP
+
+    This tool is used to replace strings in NIPAP, helpful when for example
+    moving many prefixes from a node to another. Instead of manually update
+    prefix nodes and descriptions, this tool replaces all instances of a string
+    with another.
+    
+    Currently the foolowing attributes are checked for strings to replace:
+    * node
+    * description
+
+    Usage: rename.py pattern replacement
+
+    When the script is executed, it begins by fetching all prefixes from NIPAP
+    which matches the pattern. It then iterates over them and prints out a
+    "diff", showing what lines matches and what they would be changed to. Above
+    each entry a number is displayed which identifies the entry.
+
+    Then the user is asked to select what changes to perform. The selection is
+    done by entering the numbers identifying the entries to change. Three types
+    of input is supported:
+    * The string 'all' - perform all listed replacements
+    * A comma separated list of numbers (eg. 5,7,12,37).
+      Replacement will be performed on the listed entries.
+    * A comma separated list of numbers prefixed with an exclamation mark (!, eg
+      !5,7,12,37). Replacement will be performed on all entries EXCEPT the
+      listed entries.
+"""
+
+import ConfigParser
+import os
+import time
+import re
+import sys
+
+from pynipap import Prefix, Pool, VRF
+import pynipap
+
+BATCH_SIZE = 100
+
+def replace(pattern, replacement):
+
+    # can we use colors?
+    use_colors = sys.stdout.isatty()
+
+    # Fetch prefixes matching the string to replace
+    print "Fetching prefixes from NIPAP... ",
+    sys.stdout.flush()
+    n = 1
+    prefix_list = []
+    t0 = time.time()
+    query = {
+        'operator': 'or',
+        'val1': {
+            'operator': 'regex_match',
+            'val1': 'description',
+            'val2': pattern
+        },
+        'val2': {
+            'operator': 'regex_match',
+            'val1': 'node',
+            'val2': pattern
+        }
+    }
+    full_result = Prefix.search(query, { 'parents_depth': -1, 'max_result': BATCH_SIZE })
+    prefix_result = full_result['result']
+    prefix_list += prefix_result
+    print len(prefix_list), 
+    sys.stdout.flush()
+    while len(prefix_result) == 100:
```

Comment:
I'm getting tired of this pattern. We should build a generator out of it so it's more natural to loop over prefixes.. It's not really related to your PR though ;)